butterfree/migrations/migration.py

```diff
@@ -0,0 +1,36 @@
+"""Migration entity."""
+
+from abc import ABC, abstractmethod
+from typing import Any, Callable
+
+
+class Migration(ABC):
```

Comment:
I feel like the Migration class should have only one public method: `run` that takes as input a feature set object. All other methods should be private, so we do not expose the underlying logic of a migration attempt. `run` should start by getting the feature set schema from the database (method 1.), translating to a common schema object (method 2.), comparing schemas (method 3.), creating the query given a diff (method 4.) and running the migration (method 5) which can be wrapped by a `migration_tracker` decorator. So, I feel like this abstract class could define more methods (that I've listed here).

Also, I would call this class `DatabaseMigration` so it is clearer to others that all children classes should be database-specialized migration classes.

Reply:
I've changed the class and applied your suggested changes, the only thing that is not clear to me is related to the ```migration_tracker``` decorator. This decorator could be responsible for sending logs to S3? If it's possible, then we do not need the ```_send_logs_to_s3``` method. Thanks for your suggestions and sorry for my late reply.